src/OrchardCore.Modules/OrchardCore.Users/Workflows/Drivers/RegisterUserTaskDisplayDriver.cs

```diff
@@ -15,12 +27,24 @@ protected override void EditActivity(RegisterUserTask activity, RegisterUserTask
             model.RequireModeration = activity.RequireModeration;
         }
 
-        protected override void UpdateActivity(RegisterUserTaskViewModel model, RegisterUserTask activity)
+        public async override Task<IDisplayResult> UpdateAsync(RegisterUserTask model, IUpdateModel updater)
         {
-            activity.SendConfirmationEmail = model.SendConfirmationEmail;
-            activity.ConfirmationEmailSubject = new WorkflowExpression<string>(model.ConfirmationEmailSubject);
-            activity.ConfirmationEmailTemplate = new WorkflowExpression<string>(model.ConfirmationEmailTemplate);
-            activity.RequireModeration = model.RequireModeration;
+            var viewModel = new RegisterUserTaskViewModel();
+            if (await updater.TryUpdateModelAsync(viewModel, Prefix))
+            {
+                model.SendConfirmationEmail = viewModel.SendConfirmationEmail;
+                model.RequireModeration = viewModel.RequireModeration;
+                model.ConfirmationEmailSubject = new WorkflowExpression<string>(viewModel.ConfirmationEmailSubject);
+                model.ConfirmationEmailTemplate = new WorkflowExpression<string>(viewModel.ConfirmationEmailTemplate);
+
+                if (model.SendConfirmationEmail)
+                {
+                    updater.ModelState.AddModelError(Prefix, nameof(viewModel.ConfirmationEmailSubject), S["A value is required for {0}.", nameof(viewModel.ConfirmationEmailSubject)]);
+                    updater.ModelState.AddModelError(Prefix, nameof(viewModel.ConfirmationEmailTemplate), S["A value is required for {0}.", nameof(viewModel.ConfirmationEmailTemplate)]);
```

Comment:
you need to test here if the `ConfirmationEmailSubject` is empty or not.
Otherwise this will error every time you try and save.

Reply:
Seems we need to check both subject and template before adding the error to the model state